src/mpf.rs

```diff
@@ -119,13 +120,26 @@ impl Mpf {
         }
     }
 
-    pub fn get_str(&mut self, n_digits: i32, base: i32, exp: &mut c_long) -> String{
-        let c_str = CString::new("").unwrap();
-        let out;
-        unsafe{
-            out = CString::from_raw(__gmpf_get_str(c_str.into_raw(), exp, base, n_digits, &mut self.mpf));
+    pub fn get_str(&self, n_digits: i32, base: i32, exp: &mut c_long) -> String {
+        use std::ptr::null_mut;
+        if n_digits == 0 {
+            // Maximal significant digits requested. Let GMP compute the length
+            // and allocate the space required.
+            unsafe {
+                let p = __gmpf_get_str(null_mut(), exp, base, n_digits, &self.mpf);
+                // De-allocates the GMP string on drop.
+                GString::from_raw(p).to_str().unwrap().to_string()
+            }
+        } else {
+            let bytes = unsafe {
+                // n_digits + 2 from mpf/get_str.c.
+                let mut bytes: Vec<u8> = vec![uninitialized(); n_digits as usize + 2];
```

Comment:
This create a single uninitialized byte, then copies it `n_digits + 2` times, so you haven’t really saved any work.

I think what you want is
```rust
let mut buf: Vec<c_char> = Vec::with_capacity(n_digits as usize + 2);
let res = __gmpf_get_str(buf.as_mut_ptr(), exp, base, n_digits, &self.mpf);
CStr::from_ptr(res).to_str().unwrap().to_string()
```